clang/lib/CIR/CodeGen/CIRGenStmt.cpp

```diff
@@ -280,3 +313,77 @@ mlir::LogicalResult CIRGenFunction::emitReturnStmt(const ReturnStmt &s) {
 
   return mlir::success();
 }
+
+mlir::LogicalResult CIRGenFunction::emitForStmt(const ForStmt &s) {
+  cir::ForOp forOp;
+
+  // TODO: pass in an array of attributes.
+  auto forStmtBuilder = [&]() -> mlir::LogicalResult {
+    mlir::LogicalResult loopRes = mlir::success();
+    // Evaluate the first part before the loop.
+    if (s.getInit())
```

Comment:
Oh interesting... hrmph.  This is supposed to be in scope with the for IIRC, do we need to create a lexical scope here?

Reply:
We are creating a lexical scope. This lambda gets called from the `builder.create<ScopeOp>` call on line 377, which also creates the lexical scope and sets that as the insertion point.